File: python_project_euler/Euler2.py
#!/usr/bin/env python -tt

from itertools import takewhile

def fib():
    a, b = 1, 2
    while True:
        yield a
        a, b = b, a + b
        

fibs = takewhile(lambda x : x < 4000000, fib())
print(sum(i for i in fibs if not i % 2))

File: scala_project_euler/Euler2.scala
#!/usr/bin/env sh
exec scala "$0" "$@"
!#

def fib(a:Int = 0, b:Int = 1):Stream[Int] = {
  a #:: fib(b, a + b)
}

println(fib().filter(n => n % 2 != 0).takeWhile(n => n < 4000000).sum)

